app/app.js

```diff
@@ -0,0 +1,188 @@
+            locale = state.views.i18n.locale;
+        }
+
+        const intlProvider = new IntlProvider({locale, messages: getTranslations(locale)}, {});
+        const {intl} = intlProvider.getChildContext();
+        this.setIntl(intl);
+        return intl;
+    };
+
+    getManagedConfig = async () => {
+        try {
+            const shouldStart = await AsyncStorage.getItem(DEVICE_SECURE_CACHE_KEY);
+            if (shouldStart !== null) {
+                return shouldStart === 'true';
+            }
+        } catch (error) {
+            return false;
+        }
+        return false;
+    };
+
+    setManagedConfig = (shouldStart) => {
+        AsyncStorage.setItem(DEVICE_SECURE_CACHE_KEY, shouldStart.toString());
+    };
+
+    setStartAppFromPushNotification = (startAppFromPushNotification) => {
+        this.startAppFromPushNotification = startAppFromPushNotification;
+    };
+
+    setIsNotificationsConfigured = (isNotificationsConfigured) => {
+        this.isNotificationsConfigured = isNotificationsConfigured;
+    };
+
+    setAllowOtherServers = (allowOtherServers) => {
+        this.allowOtherServers = allowOtherServers;
+    };
+
+    setAppStarted = (appStarted) => {
+        this.appStarted = appStarted;
+    };
+
+    setEMMEnabled = (emmEnabled) => {
+        this.emmEnabled = emmEnabled;
+    };
+
+    setIntl = (intl) => {
+        this.intl = intl;
+    };
+
+    setDeviceToken = (deviceToken) => {
+        this.deviceToken = deviceToken;
+    };
+
+    setReplyNotificationData = (replyNotificationData) => {
+        this.replyNotificationData = replyNotificationData;
+    };
+
+    setInBackgroundSince = (inBackgroundSince) => {
+        this.inBackgroundSince = inBackgroundSince;
+    };
+
+    setShouldRelaunchWhenActive = (shouldRelaunchWhenActive) => {
+        this.shouldRelaunchWhenActive = shouldRelaunchWhenActive;
+    };
+
+    setNativeAppLaunched = (nativeAppLaunched) => {
+        this.nativeAppLaunched = nativeAppLaunched;
+    };
+
+    verifyManagedConfigCache = async (shouldStartCache) => {
+        // since we are caching managed device results
+        // we should verify after using the cache
+        const shouldStart = await handleManagedConfig();
+        if (shouldStartCache && !shouldStart) {
+            this.setManagedConfig(false);
+            mattermostManaged.quitApp();
+            return;
+        }
+
+        this.setManagedConfig(true);
+    };
+
+    launchApp = async () => {
+        const shouldStartCache = await this.getManagedConfig();
+        if (shouldStartCache) {
+            this.startApp();
+            this.verifyManagedConfigCache(shouldStartCache);
+            return;
+        }
+
+        const shouldStart = await handleManagedConfig();
+        if (shouldStart) {
+            this.setManagedConfig(shouldStart);
+            this.startApp();
+        }
+    };
+
+    startApp = () => {
+        // TODO: should we check if entryComponent exists?
```

Comment:
Still a concern, I've tested on cold startup through the launcher and push notification and this would not be necessary. 
 
I added this comment to check if there's another way a user can end up attempting to launchApp and not have reference to the Entry component.  Resulting in the app never starting